### PR TITLE
Validate only entryServerNodes which were added at least 2 minuts ago

### PR DESCRIPTION
**Pull request by yossipapi**

### CODE REVIEW

File: alpha/lib/model/EntryServerNode.php [hunk 1]
```diff
@@ -29,4 +29,9 @@ protected function addTrackEntryInfo($trackEventType, $description)
 	
 		TrackEntry::addTrackEntry($te);
 	}
+	
+	public function validateEntryServerNode()
```
Comment (MosheMaorKaltura) [inline alpha/lib/model/EntryServerNode.php]: This API should be abstract.

File: alpha/lib/model/LiveEntry.php [hunk 2]
```diff
@@ -620,7 +620,7 @@ private static function getCacheType()
 	 * @return bool|mixed
 	 * @throws Exception
 	 */
-	private function isCacheValid(LiveEntryServerNode $liveEntryServerNode)
+	public function isCacheValid(LiveEntryServerNode $liveEntryServerNode)
 	{
```
Comment (MosheMaorKaltura) [inline alpha/lib/model/LiveEntry.php]: Must validate liveEntryServerNode is not null

File: alpha/lib/model/LiveEntryServerNode.php [hunk 3]
```diff
@@ -128,4 +128,21 @@ private function getLiveEntry()
 		
 		return $liveEntry;
 	}
+	
+	public function validateEntryServerNode()
+	{
+		$liveEntry = entryPeer::retrieveByPK($this->getEntryId());
+		if(!$liveEntry)
+		{
+			KalturaLog::debug("Entry with id [{$this->getEntryId()}] not found, will not validate entryServerNode registered");
```
Comment (MosheMaorKaltura) [inline alpha/lib/model/LiveEntryServerNode.php]: This case is abnornal, liveentryID must be found on LiveEntryServerNode, should throw exception

File: deployment/permissions/service.entryservernode.ini [hunk 4]
```diff
@@ -14,3 +14,8 @@ permissionItem4.action = list
 permissionItem4.partnerId = 0
 permissionItem4.permissions = -5>MEDIA_SERVER_BASE, -5>PARTNER_-5_GROUP_*_PERMISSION,MEDIA_SERVER_PARTNER_LEVEL, -1>PARTNER_-1_GROUP_*_PERMISSION, PARTNER_-1_GROUP_*_PERMISSION
 
+permissionItem5.service = entryservernode
+permissionItem5.action = validateRegisteredEntryServerNode
+permissionItem5.partnerId = -1
+permissionItem5.permissions = BATCH_BASE, PARTNER_-1_GROUP_*_PERMISSION
+
```
Comment (MosheMaorKaltura) [inline deployment/permissions/service.entryservernode.ini]: Should we remove this action from livestream service?

Comment (yossipapi) [inline deployment/permissions/service.entryservernode.ini]: No not yet

File: alpha/lib/model/LiveEntryServerNode.php (hunk 3 above)
Comment (yossipapi) [inline alpha/lib/model/LiveEntryServerNode.php]: Done

File: alpha/lib/model/LiveEntry.php (hunk 2 above)
Comment (yossipapi) [inline alpha/lib/model/LiveEntry.php]: Done

File: alpha/lib/model/EntryServerNode.php (hunk 1 above)
Comment (yossipapi) [inline alpha/lib/model/EntryServerNode.php]: Done.

**Comment by MosheMaorKaltura** [comment]
Reviewed

**Comment by yossipapi** [comment]
Thanks merging

**Pull request merged**